gitops/snapshot.go

```diff
@@ -775,3 +775,23 @@ func ResetSnapshotStatusConditions(adapterClient client.Client, ctx context.Cont
 
 	return nil
 }
+
+func CopySnapshotLabelsAndAnnotation(application *applicationapiv1alpha1.Application, snapshot *applicationapiv1alpha1.Snapshot, componentName string, source metav1.ObjectMeta, destination metav1.ObjectMeta, prefix string) (*applicationapiv1alpha1.Snapshot, error) {
+
+	if snapshot.Labels == nil {
+		snapshot.Labels = make(map[string]string)
+	}
+	snapshot.Labels[SnapshotTypeLabel] = SnapshotComponentType
+	snapshot.Labels[SnapshotComponentLabel] = componentName
+	snapshot.Labels[ApplicationNameLabel] = application.Name
+
+	// Copy PAC annotations/labels from source(tested snapshot or pipelinerun) to snapshot.
+	_ = metadata.CopyLabelsWithPrefixReplacement(&source, &destination, "pipelinesascode.tekton.dev", PipelinesAsCodePrefix)
+	_ = metadata.CopyAnnotationsWithPrefixReplacement(&source, &destination, "pipelinesascode.tekton.dev", PipelinesAsCodePrefix)
+
+	// Copy labels and annotations prefixed with defined prefix
+	_ = metadata.CopyLabelsByPrefix(&source, &destination, prefix)
+	_ = metadata.CopyAnnotationsByPrefix(&source, &destination, prefix)
+
+	return snapshot, nil
```

Comment:
Nothing returns error in this function, does it even make sense to declare error in return types?

Reply:
No need for return types there, removed.